src/helics/core/mpi/MpiService.cpp

```diff
@@ -0,0 +1,314 @@
+/*
+
+Copyright (C) 2017-2018, Battelle Memorial Institute
+All rights reserved.
+
+This software was co-developed by Pacific Northwest National Laboratory, operated by the Battelle Memorial Institute; the National Renewable Energy Laboratory, operated by the Alliance for Sustainable Energy, LLC; and the Lawrence Livermore National Laboratory, operated by Lawrence Livermore National Security, LLC.
+
+*/
+
+#include "MpiService.h"
+
+namespace helics {
+
+MPI_Comm MpiService::mpiCommunicator = MPI_COMM_NULL;
+
+MpiService& MpiService::getInstance ()
+{
+    static MpiService instance;
+    instance.startService ();
+    return instance;
+}
```

Comment:
you may want this to be a shared pointer, given some of the fights I have had with the shutdown sequences on different operating systems in HELICS, if this gets used in multiple threads it is entirely possible the static variable will be deleted before the uses of it are finished, leading to termination errors and seg faults.

Reply:
If a shared pointer got added to each MpiComms object, would that prevent the instance from getting destroyed until all of the destructors for MpiComms objects finished?

Reply:
I believe so,  That is what I do in some of the similar classes.  The getInstance method won't work anymore even with a shared pointer when the threads start to close but anything that captured the pointer from earlier still has access to a valid object whereas if you just captured reference that doesn't prevent the object from getting destroyed.